org.adempiere.base/src/org/compiere/util/CLogFile.java

```diff
@@ -334,6 +333,19 @@ public synchronized void publish (LogRecord record)
 		}
 	}	//	publish
 
+	private String getPrefix()
+	{
+		try
+		{
+			return Env.parseContext(Env.getCtx(), 0, System.getProperty("org.idempiere.FileLogPrefix"), false);

```

Comment:
System.getProperty("org.idempiere.FileLogPrefix") should be read once in a static variable, i.e
```
private final static s_fileLogPrefix = System.getProperty("org.idempiere.FileLogPrefix")
```

Also, if we want to have a default:
```
private static s_fileLogPrefix;
static {
     s_fileLogPrefix = System.getProperty("org.idempiere.FileLogPrefix");
     if (Util.isEmpty(s_fileLogPrefix, true) {
          s_fileLogPrefix = "@#AD_Client_Name@ - @#AD_User_Name@ - @#AD_Session_ID@";
     }
}
```

Reply:
And we shouldn't call Env.parseContext if s_fileLogPrefix is empty or null.

Reply:
@hengsin - making it static would mean it cannot be changed during server execution?

Reply:
yeah, if you want to have the flexibility to change it during server execution, better not to make it static.

Reply:
maybe a dumb question: how to change the value as it is defined on a VM argument?

Reply:
> maybe a dumb question: how to change the value as it is defined on a VM argument?

I do it frequently for logging SQL statements - even on production servers.

For example, to log SQL statements you can pass the VM argument:
```-Dorg.idempiere.db.debug=true```

Now, if you didn't define it when you started the server (the normal case in production), I open the OSGI console:
```telnet localhost 12612```

and within the OSGI console I execute:
```setprop org.idempiere.db.debug=true```

and voilà!  SQL statements start appearing in the log
after I found what I'm looking for I simply set it back to false.

Extra tip:
The amount of SQL statements is overwhelming in a production server (and even in local dev), so there is another parameter to filter what is logged based on a regular expression, for example:
```setprop org.idempiere.db.debug.filter='(?i)(?s:.)*\bAD_ChangeLog\b(?s:.)*'```
log statements that contain the ad_changelog table

Reply:
Ah great! thanks for the tip
I only use the "org.idempiere.db.debug" locally.
Great to have be able to change that on live
Merci beaucoup :)